src/app/shared/components/course-form/course-form.component.ts

```diff
@@ -122,15 +162,41 @@ export class CourseFormComponent implements OnInit {
     this.submitted = true;
     if (this.courseForm.valid) {
```

Comment:
Well, that is 3 conditionals in depth with if() -> if() -> if() without the corresponding else statements. This is block of code is getting more complicated and far away from being clean. Please refactor it to improve quality.